Sources/SyrupCore/Generator/Renderer.swift

```diff
@@ -127,6 +127,36 @@ open class Renderer {
 		]
 		return try render(template: "Operation", asFile: true, context: context)
 	}
+    
+    func renderSubscriptions(intermediateRepresentation: IntermediateRepresentation, selectionSets: SelectionSetVisitor.Results) throws -> [String] {
+        var rendered: [String] = []
+        for subscription in intermediateRepresentation.operations.subscriptions {
+            rendered.append(try renderSubscription(subscription: subscription, intermediateRepresentation: intermediateRepresentation, subscriptionSelections: selectionSets.operation(named: subscription.name)))
+        }
+        return rendered
+    }
+    
+    func renderSubscription(subscription: IntermediateRepresentation.OperationDefinition, intermediateRepresentation: IntermediateRepresentation, subscriptionSelections: SelectionSetVisitor.Operation) throws -> String {
+        let name = "\(subscription.name)Subscription"
+        let queryString = intermediateRepresentation.fullQueryString(for: subscription).removingLeadingSpaces.removingNewLines
+        let importEnums = intermediateRepresentation.referencedEnums.isEmpty == false
+        let importInputs = intermediateRepresentation.referencedInputTypes.isEmpty == false
+        let importFragments = intermediateRepresentation.fragmentDefinitions.isEmpty == false
+        let requiresCustomEncoder = subscription.variables.contains(where: { $0.type.nestedScalar() is IntermediateRepresentation.CustomCodedScalar })
+        let context: [String: Any] = [
+            "name": name,
+            "operation": subscription,
+            "queryString": queryString,
+            "moduleName": config.project.moduleName,
+            "isQuery": true,
```

Comment:
From what I can see `isQuery` is used in the Stencil (and I'm not sure where else). Is it safe to leave this as `true` until I find out if it's needed in the Stencil or not?

Reply:
I'm not sure what `isQuery` means in this context, but a subscription operation is wrapped in a `query`, so that could make sense...

Reply:
For context, this `context` is passed to the template rendering engine. If `isQuery` isn't needed in the template file for subscriptions then we can choose not to pass it at all

Reply:
IIRC `isQuery` is used in the template to render either `Query` or `Mutation` in certain places. We can just remove this line

Reply:
Why wouldn't we want to render `Subscription` in this cases? Is it just not applicable to subscriptions?

Reply:
> IIRC `isQuery` is used in the template to render either `Query` or `Mutation` in certain places. We can just remove this line

Fred is correct here. It's what ends up being used to generate class and package names with the operation type.

I think the best bet here is to drop the boolean entirely and refactor it to use the `OperationType` in `OperationDefinition` which we're already passing to the template. Or alternatively, a pre-computed value that doesn't require conditional logic in Stencil.